packages/pathfinder/src/pathfinder.ts

```diff
@@ -248,8 +329,29 @@ export class Pathfinder {
 
         // todo implement the pathfinding logic
         // be sure to use REFERENCE_ACCOUNT_PUBLIC_KEY as the destination and source accounts
-
-        return Promise.resolve([]);
+        return server.paths(Pathfinder.REFERENCE_ACCOUNT_PUBLIC_KEY, 
+                            Pathfinder.REFERENCE_ACCOUNT_PUBLIC_KEY, 
+                            destinationAsset, destinationAmount)
+                    .call()
+                    .then((paths: any) => JSON.parse(JSON.stringify(paths)).records)
+                    .then((paths: any) => {
+                        const pathDict: any = {};
+                        paths.map((pathResult: any) => {
+                            const key = Pathfinder.makeCompositeKeyFromPath(pathResult);
+                            if (!pathDict[key]) {
+                                pathDict[key] = this.constructPathInfo(pathResult);
```

Comment:
To save cpu cycles, you should only constructPathInfo on the final patharray below.

Reply:
updated